manim/mobject/svg/text_mobject.py

```diff
@@ -594,55 +590,137 @@ def set_color_by_t2c(self, t2c=None):
         """Internally used function. Sets color for specified strings."""
         t2c = t2c if t2c else self.t2c
         for word, color in list(t2c.items()):
-            for start, end in self.find_indexes(word, self.original_text):
+            for start, end in self.find_indexes(word, self.text):
                 self.chars[start:end].set_color(color)
 
     def set_color_by_t2g(self, t2g=None):
```

Comment:
Is this function used anywhere else? If not you can deprecate it or rather remove it. Same for `set_color_by_t2c`